### PR TITLE
Add support for using Configs as CredentialSpecs in services

### PR DESCRIPTION
**Pull request by thaJeztah**
Follow up to https://github.com/moby/moby/pull/38520

Brings in API changes for https://github.com/docker/swarmkit/pull/2771

**- Description for the changelog**

```
+ Add support for using Configs as CredentialSpecs in services
```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @dperny @anshulpundir @wk8 PTAL; I think this is roughly what's needed to make it work 😅

File: docs/api/version-history.md [hunk 1]
```diff
@@ -29,6 +29,8 @@ keywords: "API, Docker, rcli, REST, documentation"
 * `GET /services/{id}` now returns `Sysctls` as part of the `ContainerSpec`.
 * `POST /services/create` now accepts `Sysctls` as part of the `ContainerSpec`.
 * `POST /services/{id}/update` now accepts `Sysctls` as part of the `ContainerSpec`.
+* `POST /services/create` now accepts `Config` as part of `ContainerSpec.Privileges.CredentialSpec`.
+* `POST //services/{id}/update` now accepts `Config` as part of `ContainerSpec.Privileges.CredentialSpec`.
```
Comment (thaJeztah) [inline docs/api/version-history.md]: whoops; double slash; updating

**Comment by dperny** [comment]
i added this commit to #38632, so all of the changes are in the same place.

File: api/server/router/swarm/cluster_routes.go [hunk 2]
```diff
@@ -270,6 +275,11 @@ func (sr *swarmRouter) updateService(ctx context.Context, w http.ResponseWriter,
 				// Sysctls for docker swarm services weren't supported before
 				// API version 1.40
 				service.TaskTemplate.ContainerSpec.Sysctls = nil
+
+				if service.TaskTemplate.ContainerSpec.Privileges != nil && service.TaskTemplate.ContainerSpec.Privileges.CredentialSpec != nil {
```
Comment (silvin-lubecki) [inline api/server/router/swarm/cluster_routes.go]: nit: extract and factorize it with createService, as the logic is not that simple?

**Comment by cpuguy83** [comment]
Should this be closed in favor of #38632?

**Comment by thaJeztah** [comment]
Yes, let's close this one, as my changes were included in the other PR

**Pull request closed without merging**